File: plugin/examples/debugtalk.go
package main

import (
	"fmt"

	plugin "github.com/httprunner/hrp/plugin"
)

func SumTwoInt(a, b int) int {
	return a + b
}

func SumInts(args ...int) int {
	var sum int
	for _, arg := range args {
		sum += arg
	}
	return sum
}

func Sum(args ...interface{}) (interface{}, error) {
	var sum float64
	for _, arg := range args {
		switch v := arg.(type) {
		case int:
			sum += float64(v)
		case float64:
			sum += v
		default:
			return nil, fmt.Errorf("unexpected type: %T", arg)
		}
	}
	return sum, nil
}

func SumTwoString(a, b string) string {
	return a + b
}

func SumStrings(s ...string) string {
	var sum string
	for _, arg := range s {
		sum += arg
	}
	return sum
}

func Concatenate(args ...interface{}) (interface{}, error) {
	var result string
	for _, arg := range args {
		result += fmt.Sprintf("%v", arg)
	}
	return result, nil
}

// register functions and build to plugin binary
func main() {
	plugin.Register("sum_ints", SumInts)
	plugin.Register("sum_two_int", SumTwoInt)
	plugin.Register("sum", Sum)
	plugin.Register("sum_two_string", SumTwoString)
	plugin.Register("sum_strings", SumStrings)
	plugin.Register("concatenate", Concatenate)
	plugin.Serve()
}

File: hrp2/examples/debugtalk.py
import logging
from typing import List

import funppy


def sum(*args):
    result = 0
    for arg in args:
        result += arg
    return result

def sum_ints(*args: List[int]) -> int:
    result = 0
    for arg in args:
        result += arg
    return result

def sum_two_int(a: int, b: int) -> int:
    return a + b

def sum_two_string(a: str, b: str) -> str:
    return a + b

def sum_strings(*args: List[str]) -> str:
    result = ""
    for arg in args:
        result += arg
    return result

def concatenate(*args: List[str]) -> str:
    result = ""
    for arg in args:
        result += str(arg)
    return result

def setup_hook_example(name):
    logging.warning("setup_hook_example")
    return f"setup_hook_example: {name}"

def teardown_hook_example(name):
    logging.warning("teardown_hook_example")
    return f"teardown_hook_example: {name}"


if __name__ == '__main__':
    funppy.register("sum", sum)
    funppy.register("sum_ints", sum_ints)
    funppy.register("concatenate", concatenate)
    funppy.register("sum_two_int", sum_two_int)
    funppy.register("sum_two_string", sum_two_string)
    funppy.register("sum_strings", sum_strings)
    funppy.register("setup_hook_example", setup_hook_example)
    funppy.register("teardown_hook_example", teardown_hook_example)
    funppy.serve()
